framework/src/controller/channels/base_channel.ts

```diff
@@ -84,6 +92,6 @@ export abstract class BaseChannel {
 	// Specified as moduleName:actionName
 	abstract invoke<T>(actionName: string, params?: object): Promise<T>;
 
-	abstract registerToBus(arg: unknown): Promise<void>;
+	abstract registerToBus(arg: unknown): void;
```

Comment:
I believe we should not change this api to sync for future extensibility. Also having `async` function is not harmfull here.